identity/src/test/java/com/android/identity/processor/FlowProcessorTest.kt

```diff
@@ -0,0 +1,207 @@
+// the state class simple name). Note that state instances are serialized and deserialized
+// across invocations, moreover if server runs on multiple nodes, flow method invocations may
+// happen across multiple nodes. Thus only data stored in the state is meaningful.
+//
+// FlowEnvironment is a way to access interfaces exposed by the server engine. If a method only
+// needs flow state and parameters to do its job (a common case), this parameter will go unused.
+// In the future we may make it optional.
+//
+// Each state must be trivially constructable and have defaults for all its fields.
+
+@CborSerializable
+@FlowState(
+    flowInterface = QuadraticSolverFlow::class,
+    path = "direct"
+)
+data class DirectQuadraticSolverState(
+    var count: Int = 0
+) {
+    companion object {
+        // Getters cannot access the state, thus they must be implemented as companion methods.
+        @FlowGetter
+        fun getName(env: FlowEnvironment): String {
+            return "Direct"
+        }
+    }
+
+    @FlowMethod
+    fun solve(env: FlowEnvironment, equation: QuadraticEquation): QuadraticSolution {
+        val det = equation.b * equation.b - 4 * equation.a * equation.c
+        val result = if (det > 0) {
+            val d = sqrt(det)
+            QuadraticSolution(
+                (-equation.b + d)/(2*equation.a),
+                (-equation.b - d)/(2*equation.a)
+            )
+        } else if (det == 0.0) {
+            QuadraticSolution(-equation.b/(2*equation.a))
+        } else {
+            QuadraticSolution()
+        }
+        count++
+        return result
+    }
+
+    @FlowMethod
+    fun getCount(env: FlowEnvironment): Int = count
+}
+
+@CborSerializable
+@FlowState(
+    flowInterface = QuadraticSolverFlow::class,
+    path = "factory"
+)
+data class SolverFactoryState(
+    var totalCount: Int = 0
+) {
+    companion object
+
+    // Returning a flow on client is done merely by returning the flow state on the server.
+    @FlowMethod
+    fun createQuadraticSolver(env: FlowEnvironment): DirectQuadraticSolverState {
+        return DirectQuadraticSolverState()
+    }
+
+    // When the sub-flow of a given type is completed using FlowBaseInterface::complete call,
+    // a join method is called for the corresponding flow type.
+    @FlowJoin
+    fun completeQuadraticSolver(env: FlowEnvironment, solver: DirectQuadraticSolverState) {
+        totalCount += solver.count
+    }
+
+    @FlowMethod
+    fun getCount(env: FlowEnvironment): Int = totalCount
+}
+
+// FlowHandlerLocal handles flow calls that executed locally. It is for use on the server,
+// for running code in the app for development, and for testing.
+fun buildLocalHandler(): FlowHandlerLocal {
+    val builder = FlowHandlerLocal.Builder()
+    DirectQuadraticSolverState.register(builder)
+    SolverFactoryState.register(builder)
+    return builder.build(ByteArray(16))
```

Comment:
`ByteArray(16)` is the encryption key, right? Think I would get a more fuzzy feeling if it wasn't all zeroes, maybe even use `key = Random.Default.nextBytes(16)` so it's 100% clear.

Reply:
OK, done.